src/cli_util.rs

```diff
@@ -940,7 +940,17 @@ impl WorkspaceCommandHelper {
 
     pub fn id_prefix_context(&self) -> &IdPrefixContext<'_> {
         self.user_repo.id_prefix_context.get_or_init(|| {
-            let context: IdPrefixContext<'_> = IdPrefixContext::new(self.user_repo.repo.as_ref());
+            let mut context: IdPrefixContext<'_> = IdPrefixContext::new(self.repo().as_ref());
+            let revset_string: String = self
+                .settings
+                .config()
+                .get_string("revsets.short-prefixes")
+                .unwrap_or_else(|_| self.settings.default_revset());
+            if !revset_string.is_empty() {
+                let disambiguation_revset = self.parse_revset(&revset_string).unwrap();
+                context =
+                    context.disambiguate_within(disambiguation_revset, Some(self.workspace_id()));
+            }
```

Comment:
Borrowing `workspace_id` here makes the unsafe cast scary because it's no longer a self-referential struct.

Perhaps, it's better to make `DisambiguationData` own `Option<WorkspaceId>`.

Reply:
Yeah, that's much better. Thanks.